src/main/java/com/boj/guidance/controller/MemberController.java

```diff
@@ -0,0 +1,82 @@
+package com.boj.guidance.controller;
+
+import com.boj.guidance.config.PasswordEncoder;
+import com.boj.guidance.dto.MemberJoinRequestDto;
+import com.boj.guidance.dto.MemberLoginRequestDto;
+import com.boj.guidance.dto.MemberResponseDto;
+import com.boj.guidance.service.MemberService;
+import com.boj.guidance.util.api.ApiResponse;
+import com.boj.guidance.util.api.ResponseCode;
+import jakarta.servlet.http.HttpSession;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.*;
+
+@Slf4j
+@RestController
+@RequestMapping("/member")
+public class MemberController {
+
+    private final MemberService memberService;
+    private final PasswordEncoder passwordEncoder;
+
+    public MemberController(MemberService memberService, PasswordEncoder passwordEncoder) {
+        this.memberService = memberService;
+        this.passwordEncoder = passwordEncoder;
+    }
```

Comment:
```suggestion
@RequiredArgsConstructor
public class MemberController {

    private final MemberService memberService;
    private final PasswordEncoder passwordEncoder;
```

의존 주입 방식의 장단점을 공부해보시면 좋을 거 같아요